Speed_Controller/src/Timer/Timer.c

```diff
@@ -1,61 +1,82 @@
 #include "Timer.h"
 
-int8_t timeoutFlag = FALSE;
+#define TRUE 1
+#define FALSE 0
 
-/** Interrupt handler for Timer 2.
+static volatile int8_t timeoutFlag = FALSE;
+
+/**
+ * Initializes a timer in one pulse mode to trigger a timing interrupt after a set interval
+ *
+ * Assumes APB2 timer clock of 36MHz
+ * 
+ * @param period Period for triggering the timing interrupt in ms (must be < 65535 / 10)
  */
-void TIM2_IRQHandler(void)
+void TimerInit(unsigned int period)
 {
-	
-	if (TIM2->SR & 0x1){
-		
-		//Set a flag for action here.
-		timeoutFlag = TRUE;
-			
-		TIM2->SR &= ~(0x1);		//reset the update interrupt flag
-	}
-	
+	RCC->APB1ENR |= RCC_APB1ENR_TIM2EN; // Enable TIM2 clock
+
+	TIM2->PSC = TIM2_PRESCALER; // Processor speed is 36MHz; set prescaler value for a clock frequency of 10kHz
+
+	TIM2->CR1 &= ~TIM_CR1_DIR;	// Set counter to upcount
+	TIM2->CR1 &= ~TIM_CR1_CKD; 	// Set the timer to use default clock division
+	TIM2->CR1 |= TIM_CR1_OPM; 	// Set the timer to use one pulse mode
+
+	TIM2->ARR = 10 * period - 1;// Set autoreload value to set the period

```

Comment:
Adjusted this to assume and use a reset value of 0 instead of 1 for the timer counter. The hardware always uses a 0 reset.